plugins/cod2/cod2.cpp

```diff
@@ -3,121 +3,111 @@
 // that can be found in the LICENSE file at the root of the
 // Mumble source tree or at <https://www.mumble.info/LICENSE>.
 
-#include "../mumble_plugin_main.h"
-
-static int fetch(float *avatar_pos, float *avatar_front, float *avatar_top, float *camera_pos, float *camera_front,
-				 float *camera_top, std::string &, std::wstring &) {
-	float viewHor, viewVer;
-	char state;
-
-	for (int i = 0; i < 3; i++)
-		avatar_pos[i] = avatar_front[i] = avatar_top[i] = camera_pos[i] = camera_front[i] = camera_top[i] = 0.0f;
-
-	bool ok;
-
-	/*
-		This plugin uses the following Variables:
-
-			Address			Type	Description
-			===================================
-			0x01516608		float	Z-Coordinate (=0 out-of-game)
-			0x0151660C		float	X-Coordinate (=0 out-of-game)
-			0x01516610		float	Y-Coordinate (=0 out-of-game)
-			0x0151A114		float	Horizontal view (degrees) (=0 out-of-game)
-			0x0151A110		float	Vertical view (degrees) (=0 out-of-game)
-			0x0096B688		byte	Magic value (0=ingame/out-of-game, 4=spectator)
-	*/
-	ok = peekProc(0x0096B688, &state, 1);
-	if (!ok)
+#include "ProcessWindows.h"
+
+#include "mumble_plugin.h"
+#include "mumble_plugin_utils.h"
+
+std::unique_ptr< ProcessWindows > process;
+
+static inline bool inGame() {
+	uint8_t state;
+	if (!process->peek(0x96B688, &state, sizeof(state))) {
 		return false;
+	}
 
-	if (state == 4)
-		return true; // If this magic value is 4 we are spectating, so switch of PA
+	// 0 = in-game/out-of-game
+	// 4 = spectator
+	return state != 4;
+}
 
-	ok = peekProc(0x01516608, avatar_pos + 2, 4) && // Z
-		 peekProc(0x0151660C, avatar_pos, 4) &&     // X
-		 peekProc(0x01516610, avatar_pos + 1, 4) && // Y
-		 peekProc(0x0151A114, &viewHor, 4) &&       // Hor
-		 peekProc(0x0151A110, &viewVer, 4);         // Ver
+static int fetch(float *avatarPos, float *avatarFront, float *avatarTop, float *cameraPos, float *cameraFront,
+				 float *cameraTop, std::string &, std::wstring &) {
+	for (uint8_t i = 0; i < 3; ++i) {
+		avatarPos[i] = avatarFront[i] = avatarTop[i] = cameraPos[i] = cameraFront[i] = cameraTop[i] = 0.0f;
+	}
 
-	if (!ok)
+	if (!inGame()) {
+		return true;
```

Comment:
Shouldn't this return `false`? :eyes:

Reply:
`fetch()` returns `false` when the plugin should unlink (i.e. in case of failure) and `true` when the plugin is still okay but there is no positional data (i.e. the player is spectating).